pyquil/compatibility/v2/api/_qam.py

```diff
@@ -0,0 +1,51 @@
+from typing import Optional, Sequence, Union
+
+from rpcq.messages import ParameterAref
+from pyquil.api._qam import QAM, QAMExecutionResult, QuantumExecutable
+
+
+class StatefulQAM(QAM):
+    _loaded_executable: Optional[QuantumExecutable]
+    _result: Optional[QAMExecutionResult]
+
+    @classmethod
+    def wrap(cls, qam: QAM) -> None:
```

Comment:
This is definitely bordering on "magic". This package is not expected to be updated with new features, etc, but it's just something to keep an eye on, especially if we encounter unforeseen issues. If it were going to be longer-lived, I'd suggest a refactor, but given the circumstance I think this is OK.

Reply:
Oh, definitely. It's exactly that, though - a layer to help migration and to never build upon.